dolphin/api/v1/pools.py

```diff
@@ -12,19 +12,92 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+import six
+from webob import exc
+
+from dolphin import db, exception
+from dolphin.api import common_utils as common
 from dolphin.api.common import wsgi
+from dolphin.api.views import pools as pool_view
+from dolphin.i18n import _
 
 
 class PoolController(wsgi.Controller):
+    def __init__(self):
+        super(PoolController, self).__init__()
+        self.search_options = ['name', 'status', 'id', 'storage_id']
+    def _is_storage_valid(self, context, pool, storage_id):
+        """verify the storage and pool association  ."""
+        try:
+            if pool.storage_id != storage_id:
+                return False
+            else:
+                db.storage_get(context, storage_id)
+        except exception.StorageNotFound:
+            msg = _("storage  '%s' not found.") % storage_id
+            raise exc.HTTPNotFound(explanation=msg)
+
+    def _get_pools_search_options(self):
+        """Return pools search options allowed ."""
+        return self.search_options
+
+    def _show(self, req, pool_id, storage_id=None):
+        ctxt = req.environ['dolphin.context']
+        try:
+            pool = db.pool_get(ctxt, pool_id)
+            if storage_id:
+                if not self._is_storage_valid(ctxt, pool, storage_id):
+                    err_msg = _('Given pool %s does not have a valid '
+                                'storage.') % pool_id
+                    raise exception.InvalidStorage(reason=err_msg)
+        except exception.PoolNotFound as e:
+            raise exc.HTTPNotFound(explanation=e.msg)
+        except exception.InvalidStorage as e:
+            raise exc.HTTPConflict(explanation=e.msg)
+        return pool_view.build_pool(pool)
+
+    def _index(self, req, storage_id=None):
+        ctxt = req.environ['dolphin.context']
+        query_params = {}
+        query_params.update(req.GET)
+        # update options  other than filters
+        sort_keys = (lambda x: [x] if x is not None else x)(query_params.get
+                                                            ('sort_key'))
+        sort_dirs = (lambda x: [x] if x is not None else x)(query_params.get
+                                                            ('sort_dir'))
+        limit = query_params.get('limit', None)
+        offset = query_params.get('offset', None)
+        marker = query_params.get('marker', None)
+        # strip out options except supported search  options
```

Comment:
it seems that similar codes from storages api, can you wrapp them.

Reply:
done, wrapped and storages.py is also modified